org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/repositories/AbapGitView.java

```diff
@@ -139,6 +156,7 @@ private void showLastSelectedElement() {
 		IProject currentProject = ProjectUtil.getActiveAdtCoreProject(this.lastSelection, null, null, null);
 		if (currentProject != this.lastProject) {
 			this.lastProject = currentProject;
+			this.repoService = null;  //reset the repository Service if project is changed
```

Comment:
Maybe instead of setting the value to null, is it possible to create the repository service instance for that project and use it afterwards?